### ISSUE TITLE
[SURE-6523] Fluentbit pod filling cgroup memory, making it impossible to schedule/delete any pods on node

### ISSUE THREAD

**Issue by gaktive** [rancher/rancher#41891]
Internal reference: [SURE-6523](https://jira.suse.com/browse/SURE-6523)
Reported in 2.6.11

Issue Description:
The worker nodes cannot schedule or terminate any pods/workloads when the fluent bit (rancher-logging-root-fluentbit ) runs on that node.
Looks like the cgroup memory is getting full, and kernel is not releasing them

Impact:
Unable to schedule any workloads and need manual intervention all the time

Troubleshooting steps:
Error from the docker system; Docker is unable to perform any activity due to the errors

Kubelet logs: `Unable to perform any container activities.`

```
2023-06-15T10:20:16.852853243Z E0615 10:20:16.852618    2760 remote_runtime.go:116] "RunPodSandbox from runtime service failed" err="rpc error: code = Unknown desc = failed to start sandbox container for pod \"check-mtr-reachability-28113730-xckb4\": Error response from daemon: mounting shm tmpfs: no space left on device"
2023-06-15T10:20:16.852878513Z E0615 10:20:16.852677    2760 kuberuntime_sandbox.go:70] "Failed to create sandbox for pod" err="rpc error: code = Unknown desc = failed to start sandbox container for pod \"check-mtr-reachability-28113730-xckb4\": Error response from daemon: mounting shm tmpfs: no space left on device" pod="monitoring-mtr-reachability/check-mtr-reachability-28113730-xckb4"
2023-06-15T10:20:16.852887719Z E0615 10:20:16.852706    2760 kuberuntime_manager.go:819] "CreatePodSandbox for pod failed" err="rpc error: code = Unknown desc = failed to start sandbox container for pod \"check-mtr-reachability-28113730-xckb4\": Error response from daemon: mounting shm tmpfs: no space left on device" pod="monitoring-mtr-reachability/check-mtr-reachability-28113730-xckb4"
```

```
Jun 15 12:21:36 dcplnx23880196 dockerd[1895]: time="2023-06-15T12:21:36.637369759+02:00" level=error msg="fcee8b9683f92d22404ecccc9da9755dd35f8b272e910ad0232cb271e9c02d4f cleanup: failed to delete container from containerd: no such container"
Jun 15 12:21:36 dcplnx23880196 dockerd[1895]: time="2023-06-15T12:21:36.637417159+02:00" level=error msg="Handler for POST /v1.41/containers/fcee8b9683f92d22404ecccc9da9755dd35f8b272e910ad0232cb271e9c02d4f/start returned error: mounting shm tmpfs: no space left on device"
```

System Logs: When the fluent bit starts, it's more pressure on the memory; thus, oom killing is happening.

```
Jun 15 12:17:12 dcplnx23880196 sudo:    nrpe : TTY=unknown ; PWD=/ ; USER=root ; COMMAND=/bin/stat --printf=%s /var/lib/rancher/fluentd/log/buffer/project.c-q6qsn_p-nkf64.buffer/buffer.q5d7aadc7b3b73ef77d67d1a24f9cbb07.log
Jun 15 12:17:12 dcplnx23880196 systemd: Started Session c3043429 of user root.

Jun 15 12:19:37 dcplnx23880196 kernel: Task in /kubepods/burstable/pode43773dd-f24c-4c98-8031-f1e260eb202e/cb5085d61108f6a2669d1b4df9a2dd4fe873f371e5690abea6df76f916bbcb80 killed as a result of limit of /kubepods/burstable/pode43773dd-f24c-4c98-8031-f1e260eb202e
Jun 15 12:19:37 dcplnx23880196 kernel: memory: usage 1572864kB, limit 1572864kB, failcnt 26066959
Jun 15 12:19:37 dcplnx23880196 kernel: memory+swap: usage 1572864kB, limit 9007199254740988kB, failcnt 0
Jun 15 12:19:37 dcplnx23880196 kernel: kmem: usage 0kB, limit 9007199254740988kB, failcnt 0
Jun 15 12:19:37 dcplnx23880196 kernel: Memory cgroup stats for /kubepods/burstable/pode43773dd-f24c-4c98-8031-f1e260eb202e: cache:0KB rss:0KB rss_huge:0KB mapped_file:0KB swap:0KB inactive_anon:0KB active_anon:0KB inactive_file:0KB active_file:0KB unevictable:0KB

Jun 15 12:19:37 dcplnx23880196 kernel: Memory cgroup out of memory: Kill process 30107 (server.rb:289) score 1971 or sacrifice child
Jun 15 12:19:37 dcplnx23880196 kernel: Killed process 28500 (ruby), UID 999, total-vm:1913172kB, anon-rss:1528536kB, file-rss:0kB, shmem-rss:0kB
```

Error at Dmesg logs: It's popping all the time.

```
[Thu Jun 15 12:14:47 2023] Memory cgroup out of memory: Kill process 28207 (event_loop) score 1973 or sacrifice child
```

Further research on the error, similar GH issues were reported where Fluentbit causes such issues by holding the logs and it leads to the memory leakage bug at kernel.

The kernel running in the worker node >> 3.10.0-1160.88.1.el7.x86_64

Refer: https://github.com/coreos/bugs/issues/2427

Other related Rancher issue: https://github.com/rancher/rancher/issues/38163

**Comment by dbason** [comment]
I don't think this is unique to fluentbit, I'm fairly sure I've dealt with this same issue when using elastic filebeats.  The best fix is to upgrade kernel version to 5.4 (https://github.com/moby/moby/issues/29638#issuecomment-513991059).  For RHEL/CentOS this will require an OOB kernel update as the packaged kernel is very old.  In the mean time the best work around is putting the drop caches command into a cronjob.

I would also recommend the customer upgrade to 2,7 and the latest version of logging,

**Comment by MKlimuszka** [comment]
Reporter requested this to be closed.
